src/minichain/state.py

```
@@ -0,0 +1,86 @@
+    def apply_transaction(self, transaction: Transaction) -> None:
+        if not transaction.verify():
+            raise StateTransitionError("Transaction signature/identity verification failed")
+
+        sender = self.get_account(transaction.sender)
+        recipient = self.get_account(transaction.recipient)
+
+        if sender.nonce != transaction.nonce:
+            raise StateTransitionError(
+                f"Nonce mismatch for sender {transaction.sender}: "
+                f"expected {sender.nonce}, got {transaction.nonce}"
+            )
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)